### PR TITLE
Allow mtPaint to use Wayland (fixes #2580) 

### LINKED ISSUE #2580: mtPaint refuses to start when using Wayland
https://github.com/wjaguar/mtPaint/issues/50

### PR DESCRIPTION
**Pull request by dimkr**

### CODE REVIEW

**Comment by 01micko** [comment]
Pure wayland I presume?

**Comment by dimkr** [comment]
> Pure wayland I presume?

Yes

**Comment by dimkr** [comment]
Maybe only the `gdk_set_allowed_backends()` call needs to be removed, because all other code is unreachable under Wayland 🤔

**Comment by 01micko** [comment]
All good. I'm woofing up a slacko64 right now with your mtpaint bump and patch and will test in my labwc install.

**Comment by dimkr** [comment]
Does `mtpaint -s` work? 🤔

**Comment by 01micko** [comment]
> Does `mtpaint -s` work? thinking

Wait a minute, just testing vanilla slacko build in qemu, then transferring binary to my labwc testbed.. stay tuned!

**Comment by dimkr** [comment]
`mtpaint -s` seems to work under X, but black screen under Wayland ... import_pixmap() needs to be replaced with something like https://github.com/emersion/grim.

**Comment by 01micko** [comment]
It works! (ish)

- `mtpaint -s` fails to capture, just black
- error message when `mtpaint` is started 

```
~$ mtpaint

(mtpaint:7876): Gtk-CRITICAL **: 21:22:23.788: gtk_box_gadget_distribute: assertion 'size >= 0' failed in GtkScrollbar
```
- at least in `labwc`,  mtpaint starts with a small gui top left at 0,0 and the 'Settings Toolbar' starts in the centre of screen. Probably a `labwc` bug.

screeny (from `grim`)
20211015_21h18m47s_grim

File: woof-code/rootfs-petbuilds/mtpaint/petbuild [hunk 1]
```diff
@@ -5,6 +5,7 @@ download() {
 build() {
     unzip -q mtpaint-3.50.09.zip
     cd mtPaint-199472ad6a4ecee6c8583fb5a504a2e99712b4fc
+    patch -p1 < ../wayland.patch
```
Comment (01micko) [inline woof-code/rootfs-petbuilds/mtpaint/petbuild:8]: Should we not apply this globally and just add it to the _gtk3_ build?

Comment (01micko) [reply woof-code/rootfs-petbuilds/mtpaint/petbuild:8]: Actually, since it works ok in X11 just disregard my last comment.

**Review by 01micko — APPROVED** [review]

**Pull request merged**